src/components/EachTodo.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import { setTodo } from "../hooks/setTodo";
+
+function EachTodo({ index, todo, todoLists, setTodoLists }) {
+  const [checkState, setCheckState] = useState(todo.checked);
+  const checkboxChanged = (e) => {
+    const updatedTodoLists = [...todoLists];
+    updatedTodoLists[index] = {
+      ...updatedTodoLists[index],
+      checked: !updatedTodoLists[index].checked,
+    };
+    setTodoLists(updatedTodoLists);
+    setCheckState((prev) => !prev);
+  };
```

Comment:
```suggestion
  const checkboxChanged = () => {
    const updatedTodoLists = [...todoLists];
    updatedTodoLists[index].checked = !updatedTodoLists[index].checked;
    setTodoLists(updatedTodoLists);
  };
```
어차피 setTodoLists로 다시 받아온다면 checked도 바뀐 채로 넘어와서 checkState는 따로 관리하지 않아도 될 것 같습니다! 또한 todoLists가 바뀌면서 rendering이 되고 checkState가 바뀌면서 다시 rendering이 될 것 같아 불필요하게 두번 렌더링이 될 것 같아요

Reply:
최적화에 더 신경쓰도록 하겠습니다!! 감사합니다 ㅎㅎ